gadopt/gplates.py

```diff
@@ -0,0 +1,449 @@
+            time (float): The geological time at which velocities are to be calculated.
+            delta_time (float): The time interval over which velocity is calculated.
+
+        Returns:
+            list of pygplates.Vector3D: Velocity vectors for all domain points.
+
+        Raises:
+            RuntimeWarning: If no plate ID is found for a given point, indicating an issue with the reconstruction model.
+        """
+        # All domain points and associated (magnitude, azimuth, inclination) velocities for the current time.
+        all_domain_points = []
+        all_velocities = []
+
+        # Partition our velocity domain features into our topological plate polygons at the current 'time'.
+        # Note: pygplates can only deal with rounded ages in million years
+        plate_partitioner = pygplates.PlatePartitioner(topology_features, rotation_model, round(time, 0))
+
+        for velocity_domain_feature in velocity_domain_features:
+
+            # A velocity domain feature usually has a single geometry but we'll assume it can be any number.
+            # Iterate over them all.
+            for velocity_domain_geometry in velocity_domain_feature.get_geometries():
+
+                for velocity_domain_point in velocity_domain_geometry.get_points():
+
+                    all_domain_points.append(velocity_domain_point)
+
+                    partitioning_plate = plate_partitioner.partition_point(velocity_domain_point)
+                    if partitioning_plate:
+
+                        # We need the newly assigned plate ID to get the equivalent stage rotation of that tectonic plate.
+                        partitioning_plate_id = partitioning_plate.get_feature().get_reconstruction_plate_id()
+
+                        # Get the stage rotation of partitioning plate from 'time + delta_time' to 'time'.
+                        # Note: pygplates can only deal with rounded ages in million years
+                        equivalent_stage_rotation = rotation_model.get_rotation(round(time, 0), partitioning_plate_id, round(time, 0) + delta_time)
+
+                        # Calculate velocity at the velocity domain point.
+                        # This is from 'time + delta_time' to 'time' on the partitioning plate.
+                        # NB: velocity unit is fixed to cm/yr, but we convert it to m/yr and further on non-dimensionalise it later.
+                        velocity_vectors = pygplates.calculate_velocities(
+                            [velocity_domain_point],
+                            equivalent_stage_rotation,
+                            delta_time, velocity_units=pygplates.VelocityUnits.cms_per_yr)
+
+                        # add it to the list
+                        all_velocities.extend(velocity_vectors)
+                    else:
+                        warnings.warn("No plate id found. There is an issue with the reconstruciton model.", category=RuntimeWarning)
```

Comment:
```suggestion
                        warnings.warn("No plate id found. There is an issue with the reconstruction model.", category=RuntimeWarning)
```

If this warning does end up getting raised and you return a NaN vector, is that going to cause a weird crash somewhere else? If so, maybe it's worth raising an exception directly?

Reply:
See discussion above regarding interpolation.